src/backend/distributed/executor/local_executor.c

```diff
@@ -196,15 +196,7 @@ LocalTaskPlannedStmt(Query *workerJobQuery, Task *task, ParamListInfo boundParam
 		shardQuery = ParseQueryString(task->queryString, parameterTypes, numParams);
 	}
 
-	/*
-	 * Altough the shardQuery is local to this node, we prefer planner()
-	 * over standard_planner(). The primary reason for that is Citus itself
-	 * is not very tolarent standard_planner() calls that doesn't go through
-	 * distributed_planner() because of the way that restriction hooks are
-	 * implemented. So, let planner to call distributed_planner() which
-	 * eventually calls standard_planner().
-	 */
-	int cursorOptions = 0;
+	int cursorOptions = CURSOR_OPT_FORCE_LOCAL;
 	PlannedStmt *localPlan = planner(shardQuery, cursorOptions, boundParams);
```

Comment:
what would we need to do to safely call standard_planner here?

Reply:
I think that's a good idea. I see two options:

1) Relax the assertions in `multi_join_restriction_hook` and join_hook, so that we simply skip those when `plannerRestrictionContextList == NULL`

2) Call `CreateAndPushPlannerRestrictionContext()` and the pop functions around here.


I think (1) makes more sense